ReactSkia/textlayoutmanager/RSkTextLayoutManager.cpp

```diff
@@ -154,11 +190,47 @@ uint32_t RSkTextLayoutManager::buildParagraph (AttributedString attributedString
         fontStyle = fragment.textAttributes.fontStyle.has_value() ? 
                                  convertFontStyle(fragment.textAttributes.fontStyle.value()): 
                                  SkFontStyle::kUpright_Slant;
+        
+        fontShadowOffset = fragment.textAttributes.textShadowOffset.has_value() ?
+                                 fragment.textAttributes.textShadowOffset.value() : 
+                                 fontShadowOffset;
+
+        fontShadowRadius = (!std::isnan(fragment.textAttributes.textShadowRadius)) && (fragment.textAttributes.textShadowRadius > 0) ?
+                                 fragment.textAttributes.textShadowRadius :
+                                 fontShadowRadius;
+
+        fontLetterSpacing = !std::isnan(fragment.textAttributes.letterSpacing) ?
+                                 fragment.textAttributes.letterSpacing :
+                                 fontLetterSpacing;
+
+        fontLineHeight = (!std::isnan(fragment.textAttributes.lineHeight)) && (fragment.textAttributes.lineHeight > 0) ?
+                                 fragment.textAttributes.lineHeight :
+                                 fontLineHeight;
+
+        fontTextDecoration = fragment.textAttributes.textDecorationLineType.has_value() ?
+                                 convertDecoration(fragment.textAttributes.textDecorationLineType.value()) :
+                                 TextDecoration::kNoDecoration;  
+
+        fontShadowColor = fragment.textAttributes.textShadowColor ?
+                                 fragment.textAttributes.textShadowColor :
+                                 fragment.textAttributes.foregroundColor;
+                                 
+        
+        fontDecorationColor = fragment.textAttributes.textDecorationColor ?
+                                 fragment.textAttributes.textDecorationColor :
+                                 fragment.textAttributes.foregroundColor;
+        
+        fontLineHeightMultiplier = 1/fontSize;
 
         style.setFontSize(fontSize * fontSizeMultiplier);
         style.setFontFamilies({SkString(fragment.textAttributes.fontFamily.c_str())});
         style.setFontStyle(SkFontStyle{fontWeight, SkFontStyle::kNormal_Width, fontStyle});
-    
+        style.addShadow(convertShadow(fontShadowOffset, fontShadowRadius, convertTextPropColor(fontShadowColor)));
```

Comment:
convertShadow() can be replaced with TextShadow constructor say - TextShadow{convertTextColor(fontShadowColor),fontShadowOffset,fontShadowRadius}

Reply:
Updated